app/Policies/UserPolicy.php

```diff
@@ -37,30 +21,33 @@ public function before(?User $user, $ability)
 	 */
 	public function isAdmin(?User $user): bool
 	{
-		return $user?->id === 0;
+		return $user?->may_administrate === true;
 	}
 
 	/**
-	 * This defines if user can edit their settings.
+	 * Perform pre-authorization checks.
 	 *
-	 * @param User $user
+	 * @param User|null $user
+	 * @param string    $ability
 	 *
-	 * @return bool
+	 * @return void|bool
 	 */
-	public function canEditSettings(User $user): bool
+	public function before(?User $user, $ability)
 	{
-		return !$user->is_locked;
+		if ($this->isAdmin($user)) {
```

Comment:
As we are inside a policy, it is fine to query the capabilities of a user directly

```suggestion
		if ($user?->may_administrate === true)) {
```

This gives us the chance to remove the `isAdmin` method which does not return a right.